### PR TITLE
APR-2955 Fixed question YO-110 inactiveflow answer extraction

### PR DESCRIPTION
**Pull request by raviudari01**

### CODE REVIEW

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs [hunk 1]
```diff
@@ -224,10 +228,58 @@ public async Task<List<SubmittedApplicationAnswer>> ExtractAnswersForApplication
...
+                _logger.LogError(ex, $"Unable to extracted Partnership anwers for Application: {applicationId}");
+            }
+        }
+
+        private async Task<Answer> ExtractAnwersByQuestionTag(Guid applicationId, string questionTag, string questionId = null)
+        {
+            var answer = new Answer { QuestionId = questionId };
+
+            var questionTagData = await _qnaApiClient.GetTabularDataByTag(applicationId, questionTag);
+            if (questionTagData != null)
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs:271]: I am not able to follow this, can this be simplified as... 
```
if (questionTagData == null) return null;
answer.Value = questionTagData;
return answer;
```

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs [hunk 2]
```diff
@@ -224,10 +228,58 @@ public async Task<List<SubmittedApplicationAnswer>> ExtractAnswersForApplication
...
+            }
 
             return answers;
         }
 
+        private async Task ExtractPartnershipAnwers(Guid applicationId, List<SubmittedApplicationAnswer> answers)
+        {
+            try
+            {
+                var partnershipAnwers = await ExtractAnwersByQuestionTag(applicationId, AddPartners, QuestionIdPartnership);
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs:245]: Don't see a point of `ExtractAnwersByQuestionTag`  method as there is null check repeated here which could be avoided.

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs [hunk 3]
```diff
@@ -224,10 +228,58 @@ public async Task<List<SubmittedApplicationAnswer>> ExtractAnswersForApplication
...
+                    answers.AddRange(partnershipTabularAnswers);
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, $"Unable to extracted Partnership anwers for Application: {applicationId}");
+            }
+        }
+
+        private async Task<Answer> ExtractAnwersByQuestionTag(Guid applicationId, string questionTag, string questionId = null)
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs:266]: Spelling error in `ExtractAnwersByQuestionTag`

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs [hunk 4]
```diff
@@ -271,6 +323,7 @@ private static List<SubmittedApplicationAnswer> ExtractPageAnswers(Guid applicat
                     var submittedQuestionAnswers = ExtractQuestionAnswers(applicationId, sequenceNumber, sectionNumber, page.PageId, question, pageAnswers);
                     submittedPageAnswers.AddRange(submittedQuestionAnswers);
                 }
+
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs:326]: empty line 💥

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs [hunk 5]
```diff
@@ -330,6 +383,7 @@ private static List<SubmittedApplicationAnswer> ExtractQuestionAnswers(Guid appl
             }
             return submittedQuestionAnswers;
         }
+
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs:386]: 💥

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs [hunk 6]
```diff
@@ -224,10 +228,58 @@ public async Task<List<SubmittedApplicationAnswer>> ExtractAnswersForApplication
                     }
                 }
             }
+            //Extract QuestionId Y0-110 answer even if it is inactive, there is an issue setting answer active to false while adding orgnisation type answer to the tabular list as the last entry
+            var hasPartnershipAnswers = answers.Any(answer => answer.QuestionId == QuestionIdPartnership);
+            if(!hasPartnershipAnswers)
+            {
+                await ExtractPartnershipAnwers(applicationId, answers);
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs:235]: We only want to do this in certain case, where is that case defined?

Comment (raviudari01) [reply src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs:235]: This is the condition to check if PartnershipAnswers are already extracted or not

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs (hunk 1 above)
Comment (raviudari01) [reply src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs:273]: Refactored this

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs (hunk 3 above)
Comment (raviudari01) [reply src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs]: Corrected

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs (hunk 4 above)
Comment (raviudari01) [reply src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs]: Empty line removed

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs (hunk 5 above)
Comment (raviudari01) [reply src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs]: Empty line removed

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs (hunk 2 above)
Comment (raviudari01) [reply src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs]: Code refactored

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs [hunk 7]
```diff
@@ -224,10 +228,55 @@ public async Task<List<SubmittedApplicationAnswer>> ExtractAnswersForApplication
...
+                    answers.AddRange(partnershipTabularAnswers);
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, $"Unable to extracted Partnership answers for Application: {applicationId}");
+            }
+        }
+
+        private async Task<Answer> ExtractAnswersByQuestionTag(Guid applicationId, string questionTag, string questionId = null)
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs:268]: string questionId = null shouldn't be the default, please take off the = null

**Review by MarkFCain — APPROVED** [review]

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs (hunk 7 above)
Comment (raviudari01) [reply src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs]: Done

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs (hunk 6 above)
Comment (DevenKShah) [reply src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs]: Ravi's response below clarifies there is no need for extra checks. 
> extra check is not required, when partnership>>add people then go back and changed to sole trade route, though that answers are not cleared at partnership question, but GetAnswerByTag returns empty, so we are not adding not relevant information.

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**